### ISSUE TITLE
Flaky-test: [EndToEndMetadataTest].[testPublishConsume]

### ISSUE THREAD

**Issue by codelipenghui** [apache/pulsar#13750]
[EndToEndMetadataTest] is flaky. The [testPublishConsume] test method fails sporadically.

https://github.com/apache/pulsar/runs/4805692281?check_suite_focus=true

```
Error:  Tests run: 10, Failures: 1, Errors: 0, Skipped: 9, Time elapsed: 1.027 s <<< FAILURE! - in org.apache.pulsar.broker.EndToEndMetadataTest
Error:  testPublishConsume(org.apache.pulsar.broker.EndToEndMetadataTest)  Time elapsed: 0.166 s  <<< FAILURE!
java.lang.IllegalArgumentException: Unknown backend metadata-store
	at org.apache.bookkeeper.meta.MetadataDrivers.getBookieDriver(MetadataDrivers.java:272)
	at org.apache.bookkeeper.meta.MetadataDrivers.getBookieDriver(MetadataDrivers.java:295)
	at org.apache.bookkeeper.bookie.Bookie.instantiateMetadataDriver(Bookie.java:1141)
	at org.apache.bookkeeper.bookie.Bookie.<init>(Bookie.java:730)
	at org.apache.bookkeeper.proto.BookieServer.newBookie(BookieServer.java:152)
	at org.apache.bookkeeper.proto.BookieServer.<init>(BookieServer.java:120)
	at org.apache.pulsar.metadata.bookkeeper.BKCluster.startBookie(BKCluster.java:237)
	at org.apache.pulsar.metadata.bookkeeper.BKCluster.startNewBookie(BKCluster.java:222)
	at org.apache.pulsar.metadata.bookkeeper.BKCluster.startBKCluster(BKCluster.java:128)
	at org.apache.pulsar.metadata.bookkeeper.BKCluster.<init>(BKCluster.java:80)
	at org.apache.pulsar.broker.EmbeddedPulsarCluster.<init>(EmbeddedPulsarCluster.java:59)
	at org.apache.pulsar.broker.EmbeddedPulsarCluster.<init>(EmbeddedPulsarCluster.java:33)
	at org.apache.pulsar.broker.EmbeddedPulsarCluster$EmbeddedPulsarClusterBuilder.build(EmbeddedPulsarCluster.java:54)
	at org.apache.pulsar.broker.EndToEndMetadataTest.testPublishConsume(EndToEndMetadataTest.java:43)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by lhotari** [comment]
Similar problem:
https://github.com/apache/pulsar/runs/4811252051?check_suite_focus=true#step:10:576

**Comment by lhotari** [comment]
This seems to be related to #12770 . 
If the org.apache.bookkeeper.meta.MetadataDrivers class gets loaded before org.apache.pulsar.metadata.bookkeeper.BKCluster constructor is called, things will fail. @merlimat Could you check this problem?

https://github.com/apache/pulsar/blob/12ca27fa800c72b0c9a6e7f2583a9daa2497add9/pulsar-metadata/src/main/java/org/apache/pulsar/metadata/bookkeeper/BKCluster.java#L72-L81

https://github.com/apache/bookkeeper/blob/bc02d8c487a809fa58d75c477d9e2d5c7dedccec/bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/MetadataDrivers.java#L95-L106

**Comment by lhotari** [comment]
The workaround would be to set the system properties for all tests in pulsar-broker using the pom.xml. I can create a PR for that since it shouldn't be harmful.

**Comment by codelipenghui** [comment]
@lhotari Can we only add the properties for `EndToEndMetadataTest`, looks like other tests will not test different drivers?

**Comment by mattisonchao** [comment]
> The workaround would be to set the system properties for all tests in pulsar-broker using the pom.xml. I can create a PR for that since it shouldn't be harmful.

I think this method would be better. :)
because of 

>Another workaround would be to run EndToEndMetadataTest separately.

just a temporary solution, i think.

**Comment by lhotari** [comment]
you can see in BK code that it's not harmful:
https://github.com/apache/bookkeeper/blob/bc02d8c487a809fa58d75c477d9e2d5c7dedccec/bookkeeper-server/src/main/java/org/apache/bookkeeper/meta/MetadataDrivers.java#L95-L106

**Comment by lhotari** [comment]
You will need to close and re-open and PRs where the `CI - Unit - Brokers - Broker Group / pulsar-ci-test (group2) job` is failing to get the fix for this issue which is provided by PR #13754. 

Closing and re-opening the PR will cause the PR to create a new merge commit internally with the latest master branch changes. This is an alternative to rebasing the PR or making an explicit merge to master in the PR commits.
